app/signup/_components/SignUpForm.tsx

```diff
@@ -0,0 +1,50 @@
+"use client";
+
+import { redirect } from "next/navigation";
+import { useEffect } from "react";
+import { useFormState } from "react-dom";
+import { signup } from "~/actions/auth";
+import { Button } from "~/components/ui/button";
+import { Input } from "~/components/ui/input";
+import { Label } from "~/components/ui/label";
+
+const SignUpForm = () => {
+  const initialState = { error: null, success: false };
+  const [formState, formAction] = useFormState(signup, initialState);
+
+  useEffect(() => {
+    if (formState.success) {
+      // redirect to sign in page
+      redirect("/signin");
+    }
+  }, [formState]);
+
+  return (
+    <form action={formAction}>
+      {formState.error && (
+        <div className="text-red-500 text-center text-sm">
+          {formState.error}
```

Comment:
Can we just display `error.message` here?

Reply:
This is not a javascript Error object, this is a custom `error` data being returned from the server action and its type is `string`, so I was showing it here, I followed this method to implement that way: https://youtu.be/1cBhgW2vtZE?si=Mlt80egIM4cXhxav

if you have seen JSON object as an error message that's because I was returning it as a string from the action. I fixed it and now `formState.error` is always an error message
Commit: https://github.com/complexdatacollective/Studio/pull/13/commits/9f7eb59bc0bc40ee93900ba04b78cc7f35906b22